clang_delta/ReplaceArrayAccessWithIndex.cpp

```diff
@@ -0,0 +1,137 @@
+
+#if HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#include "ReplaceArrayAccessWithIndex.h"
+
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/Basic/SourceManager.h"
+
+#include "TransformationManager.h"
+
+#include <iostream>
+
+using namespace clang;
+using namespace llvm;
+
+
+static const char *Description =
+  "Replace array accesses with the index expression.";
+
+static RegisterTransformation<ReplaceArrayAccessWithIndex>
+Trans("replace-array-access-with-index", Description);
+
+struct ReplaceArrayAccessWithIndex::IndexCollector
+  : public RecursiveASTVisitor<ReplaceArrayAccessWithIndex::IndexCollector>
+{
+  IndexCollector(ReplaceArrayAccessWithIndex *instance);
+  bool VisitArraySubscriptExpr(ArraySubscriptExpr *ASE);
+
+  const VarDecl *getVarDeclFromExpr(const Expr *E);
+
+  ReplaceArrayAccessWithIndex *ConsumerInstance;
```

Comment:
Sorry for being pedantic - can we make it ```private``` by following the existing convention? Thanks.